src/components/ModelingMachineProvider.tsx

```diff
@@ -658,6 +662,19 @@ export const ModelingMachineProvider = ({
     editorManager.selectionRanges = modelingState.context.selectionRanges
   }, [modelingState.context.selectionRanges])
 
+  useEffect(() => {
+    const offlineCallback = () => {
+      // If we are in sketch mode we need to exit it.
+      // TODO: how do i check if we are in a sketch mode, I only want to call
```

Comment:
@Irev-Dev @franknoirot I feel like there is a way to do this so just let me know

Reply:
on second thought if we do this the useeffect will refresh on state changes and i dont think thats worth it

Reply:
`state.matches('Sketch')`